### PR TITLE
Simplify ensemble code

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: docs/src/tutorials/ensemble_modeling.md [hunk 1]
```diff
@@ -170,11 +166,9 @@ Let's see how each of our models in the ensemble compare against the data when c
...
 
 ```@example ensemble
-function prob_func(prob, i, repeat)
-    remake(probs[i]; p = fitparams[i], tspan = (t_train[1],t_ensem[end]))
-end
-fit_enprob = EnsembleProblem(prob; prob_func)
-sol = solve(enprob; trajectories=3);
+fit_probs = [remake(probs[i]; p = fitparams[i], tspan = (t_train[1],t_ensem[end])) for i in 1:3]
+fit_enprob = EnsembleProblem(fit_probs; prob_func)
+sol = solve(enprob);
```
Comment (AnasAbdelR) [inline docs/src/tutorials/ensemble_modeling.md:171]: Shouldn't this be `fit_enprob`

File: docs/src/tutorials/ensemble_modeling.md [hunk 2]
```diff
@@ -196,7 +190,7 @@ Now let's train the ensemble model. We will do that by solving a bit further tha
 calibration step. Let's build that solution data:
 
 ```@example ensemble
-sol = solve(enprob; trajectories=3);
+sol = solve(enprob);
```
Comment (AnasAbdelR) [inline docs/src/tutorials/ensemble_modeling.md:193]: This should also be `fit_enprob`

File: docs/src/tutorials/ensemble_modeling.md [hunk 3]
```diff
@@ -205,22 +199,22 @@ We can obtain the optimal weights for ensembling by solving a linear regression
...
+sol = solve(fit_enprob; saveat = t_ensem);
 ensem_prediction = sum(stack([ensem_weights[i] * sol[i][S] for i in 1:3]), dims = 2)
 plot(sol; idxs = S)
 plot!(t_ensem, ensem_prediction, lw=3)
 scatter!(t_ensem,data_ensem[1][2])
 ```
 
 ```@example ensemble
-sol = solve(enprob; saveat = t_ensem, trajectories=3);
+sol = solve(enprob; saveat = t_ensem);
```
Comment (AnasAbdelR) [inline docs/src/tutorials/ensemble_modeling.md:217]: Same thing here: `fit_enprob`

File: docs/src/tutorials/ensemble_modeling.md (hunk 2 above)
Comment (ChrisRackauckas) [inline docs/src/tutorials/ensemble_modeling.md:193]:
```suggestion
sol = solve(fit_enprob);
```

File: docs/src/tutorials/ensemble_modeling.md (hunk 3 above)
Comment (ChrisRackauckas) [inline docs/src/tutorials/ensemble_modeling.md:217]:
```suggestion
sol = solve(fit_enprob; saveat = t_ensem);
```

File: test/ensemble.jl [hunk 4]
```diff
@@ -50,14 +50,9 @@ eqs = [∂(S) ~ -β * c * I_total / N * S - v * Sv,
 @named sys3 = ODESystem(eqs)
 sys3 = structural_simplify(sys3)
 prob3 = ODEProblem(sys3, [], tspan);
+enprob = EnsembleProblem(probs)
```
Comment (ChrisRackauckas) [inline test/ensemble.jl:53]:
```suggestion
enprob = EnsembleProblem([prob, prob2, prob3])
```

File: test/ensemble.jl [hunk 5]
```diff
@@ -50,14 +50,9 @@ eqs = [∂(S) ~ -β * c * I_total / N * S - v * Sv,
...
+enprob = EnsembleProblem([prob, prob2, prob3])
 
-probs = [prob, prob2, prob3]
-function prob_func(prob, i, repeat)
-    remake(probs[i])
-end
-enprob = EnsembleProblem(prob; prob_func)
-
-sol = solve(enprob; saveat = 1, trajectories = 3);
+sol = solve(enprob; saveat = 1, trajectories=3);
```
Comment (ChrisRackauckas) [inline test/ensemble.jl:55]:
```suggestion
sol = solve(enprob; saveat = 1);
```

File: docs/src/tutorials/ensemble_modeling.md [hunk 6]
```diff
@@ -232,13 +227,11 @@ scatter!(t_ensem, data_ensem[2][2])
 Once we have obtained the ensemble model, we can forecast ahead with it:
 
 ```@example ensemble
-function prob_func(prob, i, repeat)
-    remake(probs[i]; p = fitparams[i], tspan = (t_train[1], t_forecast[end]))
-end
-fit_enprob = EnsembleProblem(prob; prob_func)
-sol = solve(enprob; trajectories = 3);
+forecast_probs = [remake(probs[i]; p = fitparams[i], tspan = (t_train[1],t_forecast[end])) for i in 1:3]
+fit_enprob = EnsembleProblem(forecast_probs; prob_func)
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/ensemble_modeling.md:231]:
```suggestion
fit_enprob = EnsembleProblem(forecast_probs)
```

File: docs/src/tutorials/ensemble_modeling.md [hunk 7]
```diff
@@ -170,11 +166,10 @@ Let's see how each of our models in the ensemble compare against the data when c
...
 
 ```@example ensemble
-function prob_func(prob, i, repeat)
-    remake(probs[i]; p = fitparams[i], tspan = (t_train[1], t_ensem[end]))
-end
-fit_enprob = EnsembleProblem(prob; prob_func)
-sol = solve(enprob; trajectories = 3);
+fit_enprob = EnsembleProblem(fit_probs)
+fit_probs = [remake(probs[i]; p = fitparams[i], tspan = (t_train[1],t_ensem[end])) for i in 1:3]
+fit_enprob = EnsembleProblem(fit_probs; prob_func)
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/ensemble_modeling.md:171]:
```suggestion
fit_probs = [remake(probs[i]; p = fitparams[i], tspan = (t_train[1],t_ensem[end])) for i in 1:3]
fit_enprob = EnsembleProblem(fit_probs)
```

File: docs/src/tutorials/ensemble_modeling.md [hunk 8]
```diff
@@ -80,18 +80,14 @@ prototype problem, which we are effectively ignoring for our use case.
 Thus a simple `EnsembleProblem` which ensembles the three models built above is as follows:
 
 ```@example ensemble
-probs = [prob, prob2, prob3]
-function prob_func(prob, i, repeat)
-    remake(probs[i])
-end
-enprob = EnsembleProblem(prob; prob_func)
+enprob = EnsembleProblem(probs; prob_func)
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/ensemble_modeling.md:83]:
```suggestion
probs = [prob, prob2, prob3]
enprob = EnsembleProblem(probs)
```

**Pull request merged**